app/controllers/additional_details/questionnaires_controller.rb

```diff
@@ -26,7 +26,7 @@ def create
   end
 
   def update
-    @questionnaire.update_attributes(questionnaire_params)
+    @questionnaire.attributes = questionnaire_params
     if @questionnaire.save
```

Comment:
I think what you're wanting to do is `if @questionnaire.update(questionnaire_params)`. 
`attributes =` will call `assign_attributes` without any sanitation of the parameters, so it's possible to bypass mass-assignment security.